app/cdap/components/RecipeManagement/CreateRecipe/index.tsx

```diff
@@ -0,0 +1,194 @@
+    formData: IRecipeFormData = recipeFormData,
+    nameErrorData: IRecipeNameErrorData = createRecipeState.recipeNameErrorData
+  ) => {
+    if (
+      formData.recipeName === '' ||
+      formData.description === '' ||
+      formData.recipeName?.trim().length === 0 ||
+      formData.description?.trim().length === 0 ||
+      nameErrorData.isRecipeNameError
+    ) {
+      setIsSaveDisabled(true);
+    } else {
+      setIsSaveDisabled(false);
+    }
+  };
+
+  const handleRecipeFormData = (formData: IRecipeFormData) => {
+    setRecipeFormData(formData);
+    handleSaveButtonMode(formData);
+  };
+
+  const onRecipeNameChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    handleRecipeFormData({
+      ...recipeFormData,
+      recipeName: event.target.value,
+    });
+    validateIfRecipeNameExists.current({
+      recipeName: event.target.value,
+      description: recipeFormData.description,
+    });
+  };
+
+  const onRecipeDescriptionChange = (event: React.ChangeEvent<HTMLTextAreaElement>) => {
+    handleRecipeFormData({
+      ...recipeFormData,
+      description: event.target.value,
+    });
+  };
+
+  const onFormSubmit = (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    onRecipeDataSave(recipeFormData);
+  };
+
+  const onRecipeDataSave = (recipeFormData: IRecipeFormData) => {
+    const requestBody = {
+      recipeName: recipeFormData.recipeName,
+      description: recipeFormData.description,
+      directives: recipeSteps,
+    };
+    createRecipeService({ requestBody, onCreateRecipeResponse, onCreateRecipeError });
+  };
+
+  const onCreateRecipeResponse = () => {
+    setShowRecipeForm(false);
+    setSnackbar({
+      open: true,
+      isSuccess: true,
+      message: `${recipeSteps.length} ${T.translate(`${PREFIX}.recipeSaveSuccessMessage`)}`,
+    });
+  };
+
+  const onCreateRecipeError = (err: Record<string, unknown>) => {
+    setShowRecipeForm(false);
+    setSnackbar({
+      open: true,
+      isSuccess: false,
+      message: (err.response as Record<string, string>).message,
+    });
+  };
+
+  const onCancel = () => {
+    setShowRecipeForm(false);
+  };
+
+  const onGetRecipeByNameError = (err: Record<string, unknown>, formData: IRecipeFormData) => {
+    if (err.statusCode === 404) {
+      setRecipeNameErrorData(noErrorState, formData);
+    }
+  };
+
+  const onGetRecipeByNameResponse = (formData: IRecipeFormData) => {
+    !createRecipeState.recipeNameErrorData.isRecipeNameError &&
+      setRecipeNameErrorData(
+        {
+          isRecipeNameError: true,
+          recipeNameErrorMessage: T.translate(`${PREFIX}.sameNameErrorMessage`).toString(),
+        },
+        formData
+      );
+  };
+
+  /*
+  In this function we are validating recipe name input filed 
+  (whether recipe name already exists or not and recipe name without alphanumeric characters) 
+  based on the result we are showing the helper text
+  */
+
+  const validateIfRecipeNameExists = useRef(
```

Comment:
Why is this a ref?

Reply:
to store the debounced function in ref, so that it can persist across renders and not get recreated on each render